bfvmm/include/intrinsics/intrinsics_intel_x64.h

```diff
@@ -79,8 +84,18 @@ class intrinsics_intel_x64 : public intrinsics_x64
     virtual bool vmread(uint64_t field, uint64_t *val)
     { return __vmread(field, val); }
 
+    virtual uint64_t vmread(uint64_t field)
```

Comment:
This function should be removed. You should be checking the returned bool for an error. Likely, we will actually add the exception logic to the C++ wrapper for the intrinsics so that it throws, in which case this might be easier in the future anyways, but yeah, the code you have here ignores the fact that a vmread could fail.